### PR TITLE
Gera.gesture selection

### PR DESCRIPTION
**Pull request by agerace**

### CODE REVIEW

File: PictureViewMaster/PictureMasterViewController.swift [hunk 1]
```diff
@@ -8,7 +8,21 @@
 
 import UIKit
 
-class PictureMasterViewController: UIViewController , UIGestureRecognizerDelegate {
+public class PictureMasterViewController: UIViewController , UIGestureRecognizerDelegate {
+    public struct Gestures : OptionSetType {
```
Comment (volbap) [inline PictureViewMaster/PictureMasterViewController.swift:12]: Isn't `public` redundant here?

Comment (agerace) [inline PictureViewMaster/PictureMasterViewController.swift:12]: Yep, but the compiler will complain if i didn't set the struct as public ☹️

**Comment by imendi** [comment]
@agerace this branch has conflicts

**Comment by agerace** [comment]
@ignaciom Yep, i'll solve them :)

File: PictureViewMaster/PictureMasterView/PictureMasterImageView.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+//  PictureMasterImageView.swift
+//  PictureViewMaster
+//
+//  Created by El gera de la gente on 4/6/16.
+//  Copyright © 2016 Inaka. All rights reserved.
+//
+
+import UIKit
+
+@objc protocol MasterViewDelegate: NSObjectProtocol {
```
Comment (volbap) [inline PictureViewMaster/PictureMasterView/PictureMasterImageView.swift]: It should be renamed to `PictureMasterImageViewDelegate`. Right now it sounds confusing about where it comes from.

File: PictureViewMaster/PictureMasterView/PictureMasterImageView.swift [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    func showImageInMasterView (image: UIImage)
```
Comment (volbap) [inline PictureViewMaster/PictureMasterView/PictureMasterImageView.swift]: To follow normal conventions when developing delegates, a better fit would be `func pictureMasterImageViewDidReceiveTap(imageView: PictureMasterImageView)` where you pass in `self` when calling it, and then you'd use `imageView.image` to retrieve the image in the class implementing this delegate.

File: PictureViewMaster/PictureMasterView/PictureMasterImageView.swift [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    }
+    
+    override init (image: UIImage?) {
+        super.init(image: image)
+        self.addTapGesture()
+    }
+    
+    private func addTapGesture (){
+        self.userInteractionEnabled = true
+        self.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(showImageInMasterView(_:))))
```
Comment (volbap) [inline PictureViewMaster/PictureMasterView/PictureMasterImageView.swift]: The `showImage` action is something that's up to the class implementing the delegate. Let's rather remove all that nomenclature from here and just name these methods with what they are here for, which is the detection of the tap.

File: README.md [hunk 5]
```diff
@@ -1 +1,69 @@
...
+[![Build Status](https://api.travis-ci.org/inaka/EventSource.svg)](https://travis-ci.org/inaka/EventSource)
+
+```
+This library supports Swift 2.3
+Swift 2.3 is currently in branch master.
+```
+
+### Abstract
+
+This is a library that allows UIImageViews (By subclassing them on the PictureMasterImageView class) to handle a touch that will enlarge the image and let the user play with them in the ways you enable. (Dragging, rotating, zooming)
```
Comment (volbap) [inline README.md]: In my opinion, you're giving too much detail in the abstract. You rather have to "sell your product" here by going to the point without entering into too many implementation details.
Example: "UIImageView subclass that adds functionality to drag, rotate and zoom it."

File: README.md [hunk 6]
```diff
@@ -1 +1,69 @@
...
+
+#### USAGE:
+
+```swift
+//Asign the delegate to the PictureMasterImageView object
+let sampleImage : PictureMasterImageView = PictureMasterImageView(image: UIImage(named:"sampleImage"))
+sampleImage.delegate = self
+
+//Then conform the the MasterViewDelegate
+func showImageInMasterView(image: UIImage) {
```
Comment (volbap) [inline README.md]: Remember to update the readme once you've changed the code with the PR comments.

File: README.md [hunk 7]
```diff
@@ -1 +1,70 @@
-# PictureViewMaster
\ No newline at end of file
+## PictureViewMaster
+Interactive image projector.
+
+
+[![Build Status](https://api.travis-ci.org/inaka/EventSource.svg)](https://travis-ci.org/inaka/EventSource)
```
Comment (volbap) [inline README.md]: Remove this line, since this is something from the `EventSource` project

File: README.md [hunk 8]
```diff
@@ -1 +1,68 @@
...
+This is a library that provides an UIImageView subclass with the capability of enlarge its image so the user can rotate it, drag it and zoom it.
+
+### How to use it?
+
+Instead of using regulars UIImageView objects use PictureMasterImageView. They'll automatically add a gesture recognizer that will let its delegate knows when the user tap on it. Then, on the delegate callback, implement the PictureMasterViewController by instantiating an object and giving it the UIImage and/or the gestures you want to support (All of them by default.). It will bring a view inside your current view with the large image and all the gesture recognizers, plus the dismissing functionality.
+NOTE: PictureMasterImageView is not actually required. You can just pass the UIImage you want to the `PictureMasterViewController` and it will also works. The UIImage subclass is just to have a nice and easy way to have clickable and enlargable UIImageViews. 
+
+### Install
+
+Drag this files to your project :
```
Comment (volbap) [inline README.md]: Drag these files to your project:

**Pull request merged**